src/api/cmd/chart-repo-sync/sync.go

```diff
@@ -100,15 +108,47 @@ func sync(repoName, repoURL string) error {
 		return err
 	}
 
+	// Process 10 charts at a time
+	numWorkers := 10
+	iconJobs := make(chan chart, numWorkers)
+	chartFilesJobs := make(chan importChartFilesJob, numWorkers)
+	var wg sync.WaitGroup
+
+	log.Debugf("starting %d workers", numWorkers)
+	for i := 0; i < numWorkers; i++ {
+		wg.Add(1)
+		go importWorker(&wg, iconJobs, chartFilesJobs)
+	}
+
+	// Enqueue jobs to process chart icons
 	for _, c := range charts {
-		if err := fetchAndImportIcon(c); err != nil {
-			log.WithFields(log.Fields{"name": c.Name}).WithError(err).Error("failed to import icon")
+		iconJobs <- c
+	}
+	// Close the iconJobs channel to signal the worker pools to move on to the
+	// chart files jobs
+	close(iconJobs)
```

Comment:
I am not that familiar with GO anymore but does this mean that we wait to import the chartFiles until all the icons for all the charts have been downloaded?

Reply:
Yes, that's right. Do you think we should get the icon and the readme for the first chart at the same time? I'm not sure it makes too much of a difference tbh.

Reply:
Well, if you show the charts even without readmes and icons I do not think that it makes a difference. If you only expose in the API fully imported charts, making it fully parallel will be good.

Because I think that the implemented the former, this looks good to me.

Thanks for the explanation.

Reply:
@migmartri it sounds like what you're suggesting is an ordering issue and not necessarily the parallelisation. With this approach, it is possible that an icon is processed at the same time as a chart's readmes, it's just that we queue up all the icons before the chart readmes so this only happens when there are < 10 icons left to process.

What would be a better way of queuing the jobs? It's somewhat more complicated by the fact that we are choosing the process the latest readme for each chart first before historical ones. However, I guess a good improvement would be to queue icon and latest chart readme at the same time.